include/dali/core/geom/mat.h

```diff
@@ -286,6 +286,24 @@ struct mat {
   #endif
     return result;
   }
+
+
+  /**
+   * @return dali::mat with ones on the diagonal and zeros elsewhere
+   */
+  DALI_HOST_DEV static constexpr mat eye() {
```

Comment:
NO. We won't have retyping existing functions. You don't like the diagonal constructor? Fine, here's an alternative that does not involve retyping:
```
DALI_HOST_DEV constexpr mat(T scalar) : mat(diag(scalar)) {}

DALI_HOST_DEV static constexpr mat diag(T scalar) {
  mat m = {};
  int n = rows < cols ? rows : cols;
  for (int i = 0; i < n; i++)
    m(i, i) = scalar;
}

DALI_HOST_DEV static constexpr mat eye() { return diag(1); }
DALI_HOST_DEV static constexpr mat identity { return diag(1); }
```

Reply:
I would use the constructor if it does what we need. If the constructor changes (what is very unlikely that we the design of mat function is reworked) tests will catch this.

Reply:
Went with @mzient 's suggestion.